demos/interactive_face_detection_demo/main.cpp

```diff
@@ -59,340 +69,519 @@ bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+                const float *y_data = out_y_fc[idx].ptr<float>();
+                const float *p_data = out_p_fc[idx].ptr<float>();
+                const float *r_data = out_r_fc[idx].ptr<float>();
+
+                std::cout << "[" << idx << "] element, yaw = " << y_data[0] <<
+                             ", pitch = " << p_data[0] <<
+                             ", roll = " << r_data[0]  << std::endl;
             }
-            slog::info << "Loading device " << deviceName << slog::endl;
-            std::cout << ie.GetVersions(deviceName) << std::endl;
+        }
 
-            /** Loading extensions for the CPU device **/
-            if ((deviceName.find("CPU") != std::string::npos)) {
-#ifdef WITH_EXTENSIONS
-                ie.AddExtension(std::make_shared<Extensions::Cpu::CpuExtensions>(), "CPU");
-#endif
+        if (landmarks_enable) {
+            // Put aassert
+            for (size_t idx = 0; idx < out_landmarks.size(); idx++) {
+                const float *lm_data = out_landmarks[idx].ptr<float>();
 
-                if (!FLAGS_l.empty()) {
-                    // CPU(MKLDNN) extensions are loaded as a shared library and passed as a pointer to base extension
-                    auto extension_ptr = make_so_pointer<IExtension>(FLAGS_l);
-                    ie.AddExtension(extension_ptr, "CPU");
-                    slog::info << "CPU Extension loaded: " << FLAGS_l << slog::endl;
+                std::cout << "[" << idx << "] element, normed facial landmarks coordinates (x, y):" << std::endl;
+
+                // FIXME: extract n_lm from out_landmarks[idx]
+                int n_lm = 70;
+                for (int i_lm = 0; i_lm < n_lm / 2; ++i_lm) {
+                    float normed_x = lm_data[2 * i_lm];
+                    float normed_y = lm_data[2 * i_lm + 1];
+
+                    std::cout << normed_x << ", " << normed_y << std::endl;
                 }
-            } else if (!FLAGS_c.empty()) {
-                // Loading extensions for GPU
-                ie.SetConfig({{PluginConfigParams::KEY_CONFIG_FILE, FLAGS_c}}, "GPU");
+            }
+        }
+}
+
+void resultProcessing(cv::Mat frame,
+                      std::list<Face::Ptr> &faces,
+                      std::list<Face::Ptr> &prev_faces,
+                      std::vector<cv::Rect> &face_hub,
+                      bool age_gender_enable,
+                      std::vector<cv::Mat> &out_ages, std::vector<cv::Mat> &out_genders,
+                      bool headpose_enable,
+                      std::vector<cv::Mat> &out_y_fc,
+                      std::vector<cv::Mat> &out_p_fc,
+                      std::vector<cv::Mat> &out_r_fc,
+                      bool landmarks_enable,
+                      std::vector<cv::Mat> &out_landmarks,
+                      bool emotions_enable,
+                      std::vector<cv::Mat> &out_emotions,
+                      size_t &id,
+                      bool no_smooth) {
```

Comment:
Same comment here. Please break it to the different processing functions per model.

Reply:
Done

Reply:
Ok